File: MobilneSystemyInformatyczne/ShareWindsurfingEquipment/src/main/java/com/project/ShareWindsurfingEquipment/model/RentalReview.java
package com.project.ShareWindsurfingEquipment.model;

import javax.persistence.*;
import java.math.BigDecimal;

@Entity
@Table
public class RentalReview {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column
    private String review;

    @Column
    private BigDecimal rating;

    @OneToOne(mappedBy = "rentalReview")
    private ItemRental itemRental;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getReview() {
        return review;
    }

    public void setReview(String review) {
        this.review = review;
    }

    public BigDecimal getRating() {
        return rating;
    }

    public void setRating(BigDecimal rating) {
        this.rating = rating;
    }

    public ItemRental getItemRental() {
        return itemRental;
    }

    public void setItemRental(ItemRental itemRental) {
        this.itemRental = itemRental;
    }

    @Override
    public String toString() {
        return "RentalReview{" +
                "id=" + id +
                ", review='" + review + '\'' +
                ", rating=" + rating +
                ", itemRental=" + itemRental +
                '}';
    }
}

File: MSI-project/src/app/models/rentalReview.ts
import { ItemRental } from "./itemRental";

export class RentalReview {

  constructor(
    private _id: number,
    private _review: string,
    private _rating: number,
    private _itemRental: ItemRental
  ){}

  public get itemRental(): ItemRental {
    return this._itemRental;
  }
  public set itemRental(value: ItemRental) {
    this._itemRental = value;
  }
  public get rating(): number {
    return this._rating;
  }
  public set rating(value: number) {
    this._rating = value;
  }
  public get review(): string {
    return this._review;
  }
  public set review(value: string) {
    this._review = value;
  }
  public get id(): number {
    return this._id;
  }
  public set id(value: number) {
    this._id = value;
  }
}
